Pythonwin/win32uimodule.cpp

```diff
@@ -2614,7 +2614,16 @@ extern "C" PYW_EXPORT BOOL Win32uiApplicationInit(Win32uiHostGlue *pGlue, TCHAR
         PySys_SetArgv(__argc - 1, __targv + 1);
 #else
     PyInit_win32ui();
-    if (argv == NULL || !PyList_Check(argv) || !PyList_Size(argv)) {
+    // Decide if we render sys.argv from command line.
+    // PY3.6- Py_Initialize sets sys.argv=NULL .
+    // PY3.7 Py_Initialize or intentional script triggers set sys.argv=[] .
+    // PY3.8+ Py_Initialize sets sys.argv=[''] - cannot be distinguished
```

Comment:
sys.argv being exactly `['']` seems unlikely to ever be a "real" argv?

Reply:
Its the normal case when plain python is started without arguments. In 3.8+ now they set that already in Py_Initialize for considering commandline (or alternatives). [] instead of NULL was in 3.7 - one could think this meant "not fully initialized (as it is indeed without commandline inspection or alternative decision) but at least: existing and a list" (that would be reasonable). But they somehow went futher perhaps thinking it was less than perfect ... 
```
C:\...>python3
Python 3.7.8 (tags/v3.7.8:4b47a5b6ba, Jun 28 2020, 08:53:46) [MSC v.1916 64 bit (AMD64)] on win32

Type "help", "copyright", "credits" or "license" for more information.
>>> sys.argv
['']
```